drivers/rawexec/driver_unix.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+//go:build !windows
+
+package rawexec
+
+import (
+	"fmt"
+	"os/user"
+
+	"github.com/hashicorp/nomad/helper/users"
+	"github.com/hashicorp/nomad/plugins/drivers"
+)
+
+func (d *Driver) Validate(cfg drivers.TaskConfig) error {
+	usernameToLookup := cfg.User
+	var user *user.User
+	var err error
+
+	// Uses the current user of the client agent process
+	// if no override is given (differs from exec)
```

Comment:
This is different than `exec` but it also seems like we could push the `users.Lookup` call into the `HasValidIDs` function because it's going to be used everywhere (just with different user names).

Reply:
You are right, we are actually doing it twice, let's move it there.